service/src/main/scala/za/co/absa/logingw/rest/SecurityConfig.scala

```diff
@@ -53,6 +56,27 @@ class SecurityConfig {
   }
 
   @Bean
-  def authenticationProvider: AuthenticationProvider = new DummyAuthenticationProvider
+  def authManager(http: HttpSecurity): AuthenticationManager = {
+    val authenticationManagerBuilder = http.getSharedObject(classOf[AuthenticationManagerBuilder])
+
+    // TODO make it autowired but only if in confid AD LDAP provider is set up
+    val adLDAPConfig = ActiveDirectoryLDAPConfig(
+      "some.domain.com",
+      "ldaps://some.domain.com:636/",
+      "(samaccountname={1})"
+    )
+
+    // TODO: take which providers and in which order to use from config
+    authenticationManagerBuilder
+      // if it is not null, on auth failure infinite recursion happens
+      .parentAuthenticationManager(null)
+      .authenticationProvider(
+        new ActiveDirectoryLDAPAuthenticationProvider(adLDAPConfig)
+      )
+      .authenticationProvider(
+        new DummyAuthenticationProvider()
```

Comment:
wow, can you really chain the providers this way?

Reply:
yup, thats cool